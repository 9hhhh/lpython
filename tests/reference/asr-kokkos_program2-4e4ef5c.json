{
    "basename": "asr-kokkos_program2-4e4ef5c",
    "cmd": "lfortran --show-asr {infile} -o {outfile}",
    "infile": "tests/kokkos_program2.f90",
    "infile_hash": "45129c6db41e13e69c776b9d135108d4a9cf7d78637044173764ae51",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-kokkos_program2-4e4ef5c.stdout",
    "stdout_hash": "689433ea3f8115317c7f7f8153d40ed5ca31564c851a7254951dab4f",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}