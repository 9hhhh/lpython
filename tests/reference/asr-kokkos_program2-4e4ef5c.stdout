(TranslationUnit (SymbolTable 49a49d05 {kokkos_program2: (Program (SymbolTable 707e323f {a: (Variable 707e323f a 0 (Real 4 [])), b: (Variable 707e323f b 0 (Real 4 [])), c: (Variable 707e323f c 0 (Real 4 [])), i: (Variable 707e323f i 0 (Integer 4 [])), nsize: (Variable 707e323f nsize 0 (Integer 4 [])), scalar: (Variable 707e323f scalar 0 (Real 4 [])), timeend: (Variable 707e323f timeend 0 (Real 4 [])), timestart: (Variable 707e323f timestart 0 (Real 4 [])), triad: (Subroutine (SymbolTable 28e5f8fd {N: (Variable 28e5f8fd N 0 (Integer 4 [])), a: (Variable 28e5f8fd a 1 (Real 4 [(() ())])), b: (Variable 28e5f8fd b 1 (Real 4 [(() ())])), c: (Variable 28e5f8fd c 2 (Real 4 [(() ())])), i: (Variable 28e5f8fd i 0 (Integer 4 [])), scalar: (Variable 28e5f8fd scalar 1 (Real 4 []))}) triad [(Var 28e5f8fd a) (Var 28e5f8fd b) (Var 28e5f8fd scalar) (Var 28e5f8fd c)] [(= (Var 28e5f8fd N) (FuncCall (Function (SymbolTable 1a7a930f {size: (Variable 1a7a930f size 4 (Integer 4 []))}) size [] [] () (Var 1a7a930f size) ()) [(Var 28e5f8fd a)] [] (Integer 4 []))) (DoConcurrentLoop ((Var 28e5f8fd i) (Num 1 (Integer 8 [])) (Var 28e5f8fd N) ()) [(= (ArrayRef (Variable 28e5f8fd c 2 (Real 4 [(() ())])) [(() (Var 28e5f8fd i) ())] (Real 4 [(() ())])) (BinOp (ArrayRef (Variable 28e5f8fd a 1 (Real 4 [(() ())])) [(() (Var 28e5f8fd i) ())] (Real 4 [(() ())])) operatorType0 (BinOp (Var 28e5f8fd scalar) operatorType2 (ArrayRef (Variable 28e5f8fd b 1 (Real 4 [(() ())])) [(() (Var 28e5f8fd i) ())] (Real 4 [(() ())])) (Real 4 [])) (Real 4 [(() ())])))])] ())}) kokkos_program2 [(= (Var 707e323f scalar) (Num 10 (Integer 8 []))) (= (Var 707e323f nsize) (FuncCall (Function (SymbolTable 1a7a930f {size: (Variable 1a7a930f size 4 (Integer 4 []))}) size [] [] () (Var 1a7a930f size) ()) [(Var 707e323f a)] [] (Integer 4 []))) (DoConcurrentLoop ((Var 707e323f i) (Num 1 (Integer 8 [])) (Var 707e323f nsize) ()) [(= (ArrayRef (Variable 707e323f a 0 (Real 4 [])) [(() (Var 707e323f i) ())] (Real 4 [])) (Num 5 (Integer 8 []))) (= (ArrayRef (Variable 707e323f b 0 (Real 4 [])) [(() (Var 707e323f i) ())] (Real 4 [])) (Num 5 (Integer 8 [])))]) (SubroutineCall 707e323f triad [(Var 707e323f a) (Var 707e323f b) (Var 707e323f scalar) (Var 707e323f c)]) (Print () [(Str ""End Stream Triad"" (Character 8 []))])]), size: (Function (SymbolTable 1a7a930f {size: (Variable 1a7a930f size 4 (Integer 4 []))}) size [] [] () (Var 1a7a930f size) ())}) [])
